### PR TITLE
feat: Create initial structure of Feast Go Operator

### PR DESCRIPTION
**Pull request by tchughesiv**
# What this PR does / why we need it:
This PR lays the groundwork for what's being proposed in the [Go Operator RFC](https://docs.google.com/document/d/1vGKMizf3_14IyiF_W_Ik7CR03joFkQfzbKT0jH4PZJM/edit).

# Which issue(s) this PR fixes:

Related to #4561

### CODE REVIEW

**Review by dmartinol — COMMENTED** [review]
👍

File: infra/feast-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+- go version v1.21.0+
+- docker version 17.03+.
+- kubectl version v1.11.3+.
+- Access to a Kubernetes v1.11.3+ cluster.
+
+### To Deploy on the cluster
+**Build and push your image to the location specified by `IMG`:**
+
+```sh
+make docker-build docker-push IMG=<some-registry>/feast-operator:tag
```
Comment (franciscojavierarceo) [inline infra/feast-operator/README.md:19]: We should update this as well.

File: infra/feast-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+**Install the CRDs into the cluster:**
+
+```sh
+make install
+```
+
+**Deploy the Manager to the cluster with the image specified by `IMG`:**
+
+```sh
+make deploy IMG=<some-registry>/feast-operator:tag
```
Comment (franciscojavierarceo) [inline infra/feast-operator/README.md:35]: same here

File: infra/feast-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+2. Using the installer
+
+Users can just run kubectl apply -f <URL for YAML BUNDLE> to install the project, i.e.:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/<org>/feast-operator/<tag or branch>/dist/install.yaml
+```
+
+## Contributing
+// TODO(user): Add detailed information on how you would like others to contribute to this project
```
Comment (franciscojavierarceo) [inline infra/feast-operator/README.md:93]: Can probably link to existing documentation

File: infra/feast-operator/config/manager/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+resources:
+- manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: feastdev/feast-operator
+  newTag: 0.40.0
```
Comment (franciscojavierarceo) [inline infra/feast-operator/config/manager/kustomization.yaml:8]: is this the latest feast release? if so it should be https://github.com/feast-dev/feast/releases/tag/v0.40.1

**Review by franciscojavierarceo — APPROVED** [review]
Left some nits that can be addressed in a follow up PR (or this one if you can)

File: infra/feast-operator/Makefile [hunk 5]
```diff
@@ -0,0 +1,322 @@
+# VERSION defines the project version for the bundle.
+# Update this value when you upgrade the version of your project.
+# To re-generate a bundle for another specific version without changing the standard setup, you can:
+# - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
+# - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
+VERSION ?= 0.40.0
```
Comment (franciscojavierarceo) [inline infra/feast-operator/Makefile:6]:
```suggestion
VERSION ?= 0.40.1
```

File: infra/feast-operator/config/manager/kustomization.yaml (hunk 4 above)
Comment (franciscojavierarceo) [reply infra/feast-operator/config/manager/kustomization.yaml:8]:
```suggestion
  newTag: 0.40.1
```

File: infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            "name": "featurestore-sample"
+          },
+          "spec": null
+        }
+      ]
+    capabilities: Basic Install
+    createdAt: "2024-10-02T20:24:54Z"
+    operators.operatorframework.io/builder: operator-sdk-v1.37.0
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v4
+  name: feast-operator.v0.40.0
```
Comment (franciscojavierarceo) [inline infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml:24]:
```suggestion
  name: feast-operator.v0.40.1
```

File: infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+  - name: Feast Operator
+    url: https://feast.dev/
+  maintainers:
+  - email: feast-discuss@googlegroups.com
+    name: feast-discuss
+  maturity: alpha
+  provider:
+    name: Feast Community
+    url: https://lf-aidata.atlassian.net/wiki/spaces/FEAST/
+  version: 0.40.0
```
Comment (franciscojavierarceo) [inline infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml:221]:
```suggestion
  version: 0.40.1
```

**Review by franciscojavierarceo — CHANGES_REQUESTED** [review]
Actually, let's fix the latest version first.

File: infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml (hunk 6 above)
Comment (tchughesiv) [reply infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml:24]: same

File: infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+                  capabilities:
+                    drop:
+                    - ALL
+              - args:
+                - --health-probe-bind-address=:8081
+                - --metrics-bind-address=127.0.0.1:8080
+                - --leader-elect
+                command:
+                - /manager
+                image: feastdev/feast-operator:0.40.0
```
[the remark this answers is not in the document]
Comment (tchughesiv) [reply infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml:132]: same

File: infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml (hunk 7 above)
Comment (tchughesiv) [reply infra/feast-operator/bundle/manifests/feast-operator.clusterserviceversion.yaml:221]: same

File: infra/feast-operator/README.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
+# feast-operator
+// TODO(user): Add simple overview of use/purpose
+
+## Description
+// TODO(user): An in-depth paragraph about your project and overview of use
```
[the remark this answers is not in the document]
Comment (tchughesiv) [reply infra/feast-operator/README.md]: tks, done

File: infra/feast-operator/README.md (hunk 1 above)
Comment (tchughesiv) [reply infra/feast-operator/README.md]: this has been moved to the contribution section of the README and i've improved this "deploy" section.

File: infra/feast-operator/README.md (hunk 2 above)
Comment (tchughesiv) [reply infra/feast-operator/README.md]: done

File: infra/feast-operator/README.md (hunk 3 above)
Comment (tchughesiv) [reply infra/feast-operator/README.md]: done

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**